lib/src/carousel/carousel.dart

```diff
@@ -0,0 +1,268 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+
+enum CarouselIndicatorType { dot, line }
+
+enum CarouselButtonType { iconOnly, rounded }
+
+class MUICarousel extends StatefulWidget {
+  /// ImageUrls for  Carousel
+  final List<String> images;
+
+  /// Height for Carousel ; default: 200
+  final double? height;
+
+  /// Duration for Carousel ; default: 5 seconds
+  final Duration? duration;
+
+  /// Border Radius for Carousel ; default: 8
+  final double? borderRadius;
+
+  /// Padding Around Carousel ; deault : All : 8
+  final EdgeInsetsGeometry? padding;
+
+  /// Indicator Type for Carousel ; default: dot
+  final CarouselIndicatorType? indicatorType;
+
+  /// Curve Animation for Carousel ; default: Curves.easeInOut
+  final Curve? curve;
+
+  /// Duration for Carousel ; default: 500 milliseconds
+  final Duration? swipeDuration;
+
+  /// Boolean to enable/disable indicator ; default: true
+  final bool? showIndicator;
+
+  /// Boolean to enable/disable indicator ; default: true
+  final bool? showButtons;
+
+  /// Carousel Button Types
+  final CarouselButtonType? buttonType;
+  const MUICarousel(
```

Comment:
Place the constructor above the variables declaration, this change was suggested by @omjogani 
Also add an optional `maxWidth` param, default value should be 430. refer to our widget creation guide for more information regarding this. 
https://utkarshs-organization-1.gitbook.io/simple-guide-to-create-modularui-widgets/

else everything LGTM 👍🏼